### DISCUSSION TITLE
How to check if neo-tree is opened?

### DISCUSSION THREAD

**Discussion by nmsobri** [nvim-neo-tree/neo-tree.nvim#826] [Q&A]
As title, is there any api i can call to check wether neo-tree is opened or not?

**Comment by cseickel** [comment]
Not really.

If you are only using the filesystem source as a sidebar, then you can chain two methods internal together to see if that particular window is open. If you use multiple sources and/or "position = current" and you want to know if any of those are open, then you would need to iterate through windows looking for one with a certain title or buffer local variable.

Which one applies to your usage?

**Reply by nmsobri** [reply]
@cseickel apparently i can use this to determine wether the sidebar is opened or not

```
  local manager = require("neo-tree.sources.manager")
  local windid = manager.get_state("filesystem").winid
```

if there neo-tree is not opened, `winid` is `nil`, otherwise it will return some integer or uniqid representing instance of neo-tree I suppose

**Reply by cseickel** [reply]
That's not entirely true. That will only tell if the sidebar has ever been open before, but it won't tell if it is currently open. If you want to know if the "filesystem" sidebar is currently open you can do this:

```lua
  local manager = require("neo-tree.sources.manager")
  local renderer = require("neo-tree.ui.renderer")

  local state = manager.get_state("filesystem")
  local window_exists = renderer.window_exists(state)
```

**Reply by nmsobri** [reply]
@cseickel ah ic.. using your code does infact return `tree-neo` status.. thx!
